.changeset/curvy-tables-kiss.md

```diff
@@ -0,0 +1,12 @@
+---
+"@fluidframework/tree": minor
+"fluid-framework": minor
+---
+---
+"section": tree
+---
+
+Fix reading of `null` from unhydrated trees
+
+Unhydrated trees containing object nodes with required fields set to `null` used to throw an error.
+This has been fixed.
```

Comment:
I think this could be more specific - is `null` explicitly allowed now, so there's no longer an error thrown?

Reply:
This is a bug fix not a feature or policy change. That is why I said "This has been fixed." as what was done about the error.

Null has always been allowed when your schema explicitly allows null somewhere. It is well supported in all other cases (all hydrated nodes, all optional fields, in arrays, when building nodes etc.)

We simply no longer have a bug which throws an error when you read a null from the tree in a specific case which is described here. I'm not sure how this could be more specific. It describes the exact case which incorrectly threw an error, and says that is fixed.

Do you have a suggestion on how this should be more specific?

Reply:
```suggestion
This was incorrect and no longer throws an error, as setting required fields to `null` is allowed in schemas where `null` is allowed elsewhere.
```
Something like this maybe? Specific may have been the wrong word to use in my original comment, but I think some elaboration would make the fix a tad clearer. I don't want to block on this though.

Reply:
Thanks for the suggestion, that clarifies where the confusion is, I'll go with a tweaked version of that.

```suggestion
This was a bug: `null` is a valid value in tree's whose schema allow it, and this specific case now correctly returns `null` values when appropriate without erroring.
```